### PR TITLE
fix redirect on Windows

### PR DESCRIPTION
**Pull request by mattn**
Simply redirecot doesn't work on Windows.
```
:echo system("echo 1>test")
```

Diff is bits large, but only add part for quoted string (like below).

```c
    if (*p_sxe != NUL && *p_sxq == '(')
    {
	if (itmp != NULL || otmp != NULL)
	    vim_snprintf((char *)buf, len, "(%s)", (char *)cmd);
	else
	    STRCPY(buf, cmd);
	if (itmp != NULL)
	{
	    STRCAT(buf, " < ");
	    STRCAT(buf, itmp);
	}
    }
    else
    {
        /* original part */
    }
```

### CODE REVIEW

File: src/testdir/test_system.vim [hunk 1]
```diff
@@ -1,25 +1,23 @@
 " Tests for system() and systemlist()
 
 function! Test_System()
-  if !executable('echo') || !executable('cat') || !executable('wc')
```
Comment (k-takata) [inline src/testdir/test_system.vim:6]: Is it okay to totally remove this?
Isn't it needed for non-windows platform?

File: src/testdir/test_system.vim [hunk 2]
```diff
@@ -1,25 +1,23 @@
...
-  if out != "123 \n"
-    call assert_equal("123\n", out)
+  if !has('win32')
+    call assert_equal("123\n", system('echo 123'))
+    call assert_equal(['123'], systemlist('echo 123'))
+    call assert_equal('123',   system('cat', '123'))
+    call assert_equal(['123'], systemlist('cat', '123'))
+    call assert_equal(["as\<NL>df"], systemlist('cat', ["as\<NL>df"]))
+  else
+    call assert_equal("123\n", system('cmd /c echo 123'))
```
Comment (k-takata) [inline src/testdir/test_system.vim]: Why is `cmd /c` used here?
For someone who sets shell to bash.exe or something?

File: src/testdir/test_system.vim [hunk 3]
```diff
@@ -1,25 +1,23 @@
...
+    call assert_equal("123\n",   system('cmd /c more', '123'))
+    call assert_equal(["123\r"], systemlist('cmd /c more', '123'))
+    call assert_equal(["as\r", "df\r"], systemlist('cmd /c more', ["as\<NL>df"]))
   endif
 
-  let out = systemlist('echo 123')
-  " On Windows we may get a trailing space and CR.
-  if out != ["123 \r"]
-    call assert_equal(['123'], out)
+  if executable('wc')
```
Comment (k-takata) [inline src/testdir/test_system.vim]: Shouldn't be `!executable('cat') || !executable('wc')`?

File: src/ex_cmds.c [hunk 4]
```diff
@@ -1821,8 +1841,10 @@ append_redir(
     }
     if (p != NULL)
     {
-	*end = ' '; /* not really needed? Not with sh, ksh or bash */
-	vim_snprintf((char *)end + 1, (size_t)(buflen - (end + 1 - buf)),
+#ifdef WIN3264
```
Comment (k-takata) [inline src/ex_cmds.c]: `MSWIN` should be used instead of `WIN3264`.

**Comment by brammool** [comment]
Looks like this is still relevant.  Can you sync to head?

**Comment by mattn** [comment]
@brammool updated.

**Pull request closed without merging**